packages/amplify_authenticator/lib/src/mixins/authenticator_date_field.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:amplify_authenticator/src/theme/amplify_theme.dart';
+import 'package:amplify_authenticator/src/widgets/form_field.dart';
+import 'package:intl/intl.dart';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+final DateFormat formatter = DateFormat('yyyy-MM-dd');
+
+mixin AuthenticatorDateField<FieldType,
+        T extends AuthenticatorFormField<FieldType, String, T>>
+    on AuthenticatorFormFieldState<FieldType, String, T> {
+  String _convertToDateString(DateTime dt) {
+    return formatter.format(dt);
+  }
+
+  late TextEditingController _controller;
+
+  @override
+  void initState() {
+    _controller = TextEditingController(text: '');
+    super.initState();
+  }
+
+  @override
+  Widget buildFormField(BuildContext context) {
+    final inputResolver = stringResolver.inputs;
+    final hintText = widget.hintText == null
+        ? inputResolver.resolve(context, widget.hintTextKey!)
+        : widget.hintText!;
+
+    DateTime now = DateTime.now();
+    Future<void> _pickTime() async {
+      final date = await showDatePicker(
+          context: context,
+          initialDate: now,
+          firstDate: DateTime(DateTime.now().year - 110),
+          lastDate: now);
+      _controller.text = _convertToDateString(date!);
```

Comment:
`date` can be null here, I'm pretty sure, if they cancel without selecting a date. Need a null check.

Reply:
Thanks, good catch.